test/unit/DetectChanges/BCBreak/MethodBased/MethodParameterAddedTest.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RoaveTest\BackwardCompatibility\DetectChanges\BCBreak\MethodBased;
+
+use PHPUnit\Framework\TestCase;
+use Roave\BackwardCompatibility\Change;
+use Roave\BackwardCompatibility\Changes;
+use Roave\BackwardCompatibility\DetectChanges\BCBreak\MethodBased\MethodBased;
+use Roave\BackwardCompatibility\DetectChanges\BCBreak\MethodBased\MethodParameterAdded;
+use Roave\BetterReflection\BetterReflection;
+use Roave\BetterReflection\Reflection\ReflectionClass;
+use Roave\BetterReflection\Reflection\ReflectionMethod;
+use Roave\BetterReflection\Reflector\DefaultReflector;
+use Roave\BetterReflection\SourceLocator\Type\StringSourceLocator;
+
+use function array_combine;
+use function array_keys;
+use function array_map;
+use function assert;
+use function iterator_to_array;
+
+final class MethodParameterAddedTest extends TestCase
+{
+    private MethodBased $methodCheck;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->methodCheck = new MethodParameterAdded();
+    }
+
+    public function testWillSkipCheckingPrivateMethods(): void
+    {
+        $from = $this->createMock(ReflectionMethod::class);
+        $to   = $this->createMock(ReflectionMethod::class);
+
+        $from
+            ->method('isPrivate')
+            ->willReturn(true);
+
+        self::assertEquals(Changes::empty(), ($this->methodCheck)($from, $to));
+    }
+
+    public function testWillSkipCheckingMethodsOnFinalClasses(): void
+    {
+        $from = $this->createMock(ReflectionMethod::class);
+        $to   = $this->createMock(ReflectionMethod::class);
+
+        $from
+            ->method('isFinal')
+            ->willReturn(true);
+
+        self::assertEquals(Changes::empty(), ($this->methodCheck)($from, $to));
+    }
+
+    /**
+     * @param string[] $expectedMessages
+     *
+     * @dataProvider methodsToBeTested
+     */
+    public function testDiffs(
+        ReflectionMethod $fromMethod,
+        ReflectionMethod $toMethod,
+        array $expectedMessages,
+    ): void {
+        $changes = (new MethodParameterAdded())($fromMethod, $toMethod);
+
+        self::assertSame(
+            $expectedMessages,
+            array_map(static function (Change $change): string {
+                return $change->__toString();
+            }, iterator_to_array($changes)),
+        );
+    }
+
+    /** @return array<string, array{0: ReflectionMethod, 1: ReflectionMethod, 2: list<string>}> */
+    public function methodsToBeTested(): array
+    {
+        $astLocator = (new BetterReflection())->astLocator();
+
+        $fromLocator = new StringSourceLocator(
+            <<<'PHP'
+<?php
+
+class TheClass {
+    public function addedParameter() {}
+}
+PHP
+            ,
+            $astLocator,
+        );
+
+        $toLocator = new StringSourceLocator(
+            <<<'PHP'
+<?php
+
+class TheClass {
+    public function addedParameter(array $options = []) {}
```

Comment:
This is a breakage because child classes will have to adapt, right?

Reply:
exactly!